docker/indexer/stages/processing/processing.go

```diff
@@ -0,0 +1,132 @@
+// Package processing implements the hashing step for each provide input.
+package processing
+
+import (
+	"context"
+	"crypto/md5"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"cloud.google.com/go/storage"
+	"github.com/go-git/go-git/v5"
+	"github.com/google/osv.dev/docker/indexer/shared"
+	"github.com/google/osv.dev/docker/indexer/stages/preparation"
+
+	log "github.com/golang/glog"
+	pb "github.com/google/osv.dev/docker/indexer/proto"
+)
+
+const workers = 25
+
+// Storer is used to permanently store the results.
+type Storer interface {
+	Store(ctx context.Context, repoInfo *preparation.Result, fileResults []FileResult) error
+}
+
+// FileResult holds the per file hash and path information.
+type FileResult struct {
+	Path string
+	Hash [md5.Size]byte
```

Comment:
Why md5 instead of say sha1? Does it make a huge difference in performance here?

Reply:
With my initial test md5 was faster to compute and with the amount of hashes that we need to compute it seemed like the right decision to use the faster one.